src/lib.rs

```diff
@@ -0,0 +1,400 @@
+                    _ => {}
+                }
+            }
+            None
+        },
+        None => None
+    }
+}
+
+fn make_statement_body(first_block: Element) -> StatementBody {
+    let mut body = StatementBody::new();
+
+    // Create each block, put them into the statement body
+    let mut block_el: Element;
+    block_el = first_block;
+    loop {
+        body.blocks.push(make_block(block_el));
+        let next_block = get_next_block_element(block_el);
+        match next_block {
+            Some(next_block_el) => {
+                block_el = next_block_el;
+            },
+            None => {
+                break;
+            }
+        }
+    }
+
+    body
+}
+
+fn make_block(block_el: Element) -> Block {
+    let mut block = Block::new();
+    for attribute in block_el.attributes().iter() {
+        let name = attribute.name().local_part();
+        let value = attribute.value().to_string();
+        match name {
+            "type" => { block.block_type = value; },
+            "id" => { block.id = value; },
+            _ => {}
+        }
+    }
+    for child in block_el.children().iter() {
+        match child {
+            &ChildOfElement::Element(child_el) => {
+                let child_name = child_el.name().local_part();
+                match child_name {
+                    "statement" => {
+                        let statement_el = child_el;
+                        let statement_name = get_attribute(statement_el, "name").unwrap();
+                        let statement_body = match get_first_child_element(statement_el) {
+                            Some(first_child_block) => make_statement_body(first_child_block),
+                            None => StatementBody::new()
+                        };
+                        block.statements.insert(statement_name, statement_body);
+                    },
+                    "field" => {
+                        let field_el = child_el;
+                        let field_name = get_attribute(field_el, "name").unwrap();
+                        let field_value = make_field_value(field_el);
+                        block.fields.insert(field_name, field_value);
+                    },
+                    _ => {}
+                }
+            },
+            _ => {}
+        }
+    }
+    block
+}
+
+fn make_field_value(field_el: Element) -> FieldValue {
+    for child in field_el.children().iter() {
+        match child {
+            &ChildOfElement::Text(text_node) => {
+                let value = text_node.text().to_string();
+                return FieldValue::SimpleField(value);
+            },
+            _ => panic!("TODO: Implement expression fields")
+        }
+    }
+    panic!("Expected child nodes for field");
+}
+
+// General DOM utilities
+
+fn get_xml_element(document: Document) -> Element {
+    let root: Root = document.root();
+    let root_children = root.children();
+    for child in root_children.iter() {
```

Comment:
As above, use `root_children.iter().filter(...).next()` and return that directly, which will be an `Option<Element>`. The function returns `None` if there is no xml element (as the panic says). If you truly want to panic, use `.expect("Cannot find xml element")` on the `.next()`.